src/classes/BDI_FieldMapping.cls

```diff
@@ -50,4 +50,5 @@ public interface BDI_FieldMapping {
             String targetObjectName, 
             List<String> dataImportFields);
 
+    Map<String, Map<String, Set<String>>> getTargetObjectsAndFieldsByDataImportField();
```

Comment:
Is it worth considering an inner class to encapsulate the Map<String, Set<String>> inner part of this collection? That would avoid the nested maps